compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -73,7 +73,9 @@ pub(super) fn opt_const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<
                         .and_then(|args| {
                             args.args
                                 .iter()
-                                .filter(|arg| arg.is_const())
+                                .filter(|arg| {
+                                    matches!(arg, GenericArg::Infer(_) | GenericArg::Const(_))
```

Comment:
this isn't quite right, further down we use the `arg_index` while filtering by `matches!(param.kind, ty::GenericParamDefKind::Const { .. })` so we can get mismatches here. I think we want to filter with `!arg.is_lifetime()`

Reply:
hm just changing it to lifetimes doesn't seem to work, is there any reason to filter later and just move the conditional into the `position` call?

Reply:
you also have to modify the `GenericParamKind` filter when searching for the param in the generics.

Reply:
i.e. use `!is_lifetime` both when computing and when using the `arg_index`

Reply:
I've been trying to finagle this, but it seems to be more annoying than this small change, so I'll just leave a `FIXME`